lib/SPIRV/PreprocessMetadata.cpp

```diff
@@ -167,11 +167,21 @@ void PreprocessMetadataBase::visit(Module *M) {
 
     // !{void (i32 addrspace(1)*)* @kernel, i32 35, i32 size}
     if (MDNode *ReqdSubgroupSize = Kernel.getMetadata(kSPIR2MD::SubgroupSize)) {
-      EM.addOp()
+      // A primary named subgroup size is encoded as 
```

Comment:
Strictly speaking, we are changing default behavior here. Right now the translator propagates sub-group size 0 as-is, but with the change applied it will only happen if the extension is enabled.

I would maybe suggest that we preserve the original behavior to avoid accidentally breaking someone's else workflow, i.e. we should only react to `0` if the extension is enabled even at this step, where we only emit some internal metadata.

Reply:
I changed the behavior and added a test case: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2317/commits/193c25d6f6b9f7f10c8b10deb7057a0050a3b1b7

Note this code is preprocessing the LLVM module and adding extra execution metadata to that module, so I changed this code to add this metadata in addition to the regular sub group metadata (as opposed to instead). Then in `SPIRVWriter`, sub group execution mode metadata will be present and be processed like before and the named sub group metadata will be ignored when the extension is turned off. https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/2317/files#diff-ba003da77a80df1db76fb378a873a80e758591d72bf82a46fc90069b2a947990R5838-R5843